qemu/lib/iozone.py

```diff
@@ -0,0 +1,146 @@
+            test_cmd = iozone_test_cmd % mountpoint
+            test_cmd = utils_misc.set_winutils_letter(self.session, test_cmd)
+        elif self.params.get("os_type") == "linux":
+            test_cmd = iozone_test_cmd % mountpoint
+            self.session.cmd_output_safe('cd %s' % self.iozone_work_path)
+
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        self.results_path = os.path.join(self.test.resultsdir, 'raw_output_%s'
+                                         % self.test.iteration)
+        logging.info("Write results to %s" % self.results_path)
+        #omit write results error
+        try:
+            with open(self.results_path, 'w') as iozone_result:
+                iozone_result.write(output)
+        except:
+            pass
+
+        return status, output
+
+    def after_run_iozone(self):
+        """
+        Clean up iozone env/post result after iozone test
+        """
+        raise NotImplementedError()
+
+
+class IozoneWin(_IozoneTest):
+    """
+    Post result after iozone test.
+    """
+    def __init__(self, vm, params, test):
+        super(IozoneWin, self).__init__(vm, params, test)
+
+    def before_run_iozone(self):
+        """
+        Do some preparations before run iozone test
+        """
+        pass
+
+    def after_run_iozone(self):
+        """
+        Post result after iozone test, generate a series graphs
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -b
+        # I:\iozone.xls -f I:\testfile".
+        # Otherwise, your result can't be analysed correctly.
+        if self.params.get("post_results", "no") == "yes":
+            logging.info("Generate graph of test result")
+            analysisdir = os.path.join(self.test.resultsdir, 'analysis_%s'
+                                       % self.test.iteration)
+            iozone_analyzer = postprocess_iozone.IOzoneAnalyzer(list_files=[self.results_path],
+                                                                output_dir=analysisdir)
+            iozone_analyzer.analyze()
+            iozone_plotter = postprocess_iozone.IOzonePlotter(results_file=self.results_path,
+                                                              output_dir=analysisdir)
+            iozone_plotter.plot_all()
+
+
+class IozoneLinux(_IozoneTest):
+    """
+    Install iozone test tool before test, clean up iozone packets after test
+    """
+    def __init__(self, vm, params, test):
+        super(IozoneLinux, self).__init__(vm, params, test)
+        stress_type = "iozone"
+        self.iozone_work_path = params.get("iozone_work_path", "/home/iozone3_458/src/current")
+        download_url = params.get("download_url_iozone",
+                                  "http://www.iozone.org/src/stable/iozone3_458.tar")
```

Comment:
Suggest using local file first, anyway, if do need to download it everytime, do not set the default value.